src/tir/contrib/ethosu/passes.cc

```diff
@@ -223,6 +224,572 @@ tvm::transform::Pass CopyComputeReordering(Optional<Integer> max_copy_movements)
+    auto buffers_to_merge{
+        get_args_to_merge_without_args_not_in_const_dict(args_to_merge, const_dict)};
+    auto new_const_dict{make_new_const_dict(buffers_to_merge, const_dict)};
+
+    // Make the new prim func
+    auto prim_func_node{main_func.CopyOnWrite()};
+    prim_func_node->body = std::move(new_body);
+    prim_func_node->buffer_map = std::move(new_buffer_map);
+    prim_func_node->params = std::move(new_params);
+    prim_func_node->preflattened_buffer_map = {};
+    PrimFunc f{GetRef<PrimFunc>(prim_func_node)};
+
+    // Add the new const dict as an attribute
+    f = WithAttr(std::move(f), "ethos-u.const-dict", new_const_dict);
+
+    return f;
+  }
+
+ private:
+  /*! Indicates whether the pass is analyzing or rewriting */
+  bool analyze = true;
+
+  /*! A stack to store allocates as they are visited. */
+  std::vector<Allocate> allocates{};
+
+  /*! A list that contains in the i-th position the write buffer of the i-th statement
+   * if that statement is a copy to a buffer with global scope  */
+  std::vector<Optional<Buffer>> copy_write_buffers{};
+
+  /*! Maps a copy's write buffer to an index representing the
+   * new buffer and an offset in that buffer */
+  std::map<Buffer, std::pair<int /* new buffer index */, int /* offset */>>
+      old_to_new_write_buffer{};
+
+  /*! Maps an index representing a new buffer to the length of that buffer */
+  std::map<int /* new buffer index */, int /* length */> new_buffers_length{};
+
+  /*! Maps an index representing a new buffer to the new buffer */
+  std::map<int /* new buffer index */, Buffer> new_buffers{};
+
+  /*! Maps an index representing a new buffer to the cycle_counts needed to copy that buffer */
+  std::map<int /* new buffer index */, int64_t> cycle_counts{};
+
+  /*! Maps a copy's read buffer to the new copy's read buffer */
+  std::map<Buffer, Buffer> old_to_new_read_buffers{};
+
+  /*! Maps an index representing a new buffer to the list of buffers to be merged in the new buffer
+   */
+  std::map<int /* new buffer index */, std::vector<Buffer>> buffers_to_merge{};
+
+  /*! A set of buffers to delete */
+  std::set<Buffer> buffers_to_delete{};
+
+  // Visit
+
+  Stmt VisitStmt_(const AllocateNode* op) override {
+    if (analyze) {
+      allocates.push_back(GetRef<Allocate>(op));
+      return VisitStmt(op->body);
+    } else {
+      auto allocate{CopyOnWrite(op)};
+      allocate->body = this->VisitStmt(op->body);
+      return Stmt(allocate);
+    }
+  }
+
+  Stmt VisitStmt_(const SeqStmtNode* op) override {
+    if (op->size() <= 1) {
+      return StmtExprMutator::VisitStmt_(op);
+    }
+    return analyze ? analyze_seq_stmt(op) : rewrite_seq_stmt(op);
+  }
+
+  Stmt analyze_seq_stmt(const SeqStmtNode* op) {
+    auto seq_stmt{GetRef<SeqStmt>(op)};
+
+    for (size_t i = 0; i < seq_stmt.size(); ++i) {
+      Stmt stmt{seq_stmt[i]};
+
+      switch (get_stmt_type(stmt)) {
```

Comment:
We need a default here.
Ref : https://google.github.io/styleguide/cppguide.html#Loops_and_Switch_Statements

Reply:
It is a switch on an enumerated value, and all cases are covered. So we don't need a default, don't we?